p2p/p2p.go

```diff
@@ -210,6 +210,49 @@ func (f peerRoutingFunc) FindPeer(ctx context.Context, p peer.ID) (peer.AddrInfo
 	return f(ctx, p)
 }
 
+// ForceDirectConnections attempts to establish a direct connection if there is an existing relay connection to the peer.
+// The idea is to enable switching to a direct connection as soon as the host has a connection to the peer.
+func ForceDirectConnections(ctx context.Context, tcpNode host.Host, peerIDs []peer.ID) {
+	ticker := time.NewTicker(1 * time.Minute)
+	forceDirectConn := func() {
+		for _, p := range peerIDs {
+			conns := tcpNode.Network().ConnsToPeer(p)
+			directConn := false
+			if len(conns) == 0 {
+				continue
+			}
+
+			for _, conn := range conns {
+				if !IsRelayAddr(conn.RemoteMultiaddr()) {
+					directConn = true
+					break
+				}
+			}
+
+			if !directConn {
+				// All existing connections are through relays, so we can try force dialing a direct connection.
+				err := tcpNode.Connect(network.WithForceDirectDial(ctx, "relay_to_direct"), peer.AddrInfo{ID: p})
+				if err != nil {
+					log.Warn(ctx, "Direct connection to peer unsuccessful", err, z.Str("peer", p.String()))
+				} else {
+					log.Debug(ctx, "Direct connection to peer successful", z.Str("peer", p.String()))
+				}
+			}
+		}
+	}
+
+	go func() {
```

Comment:
cool, adding a new `lifecycle.StartForceDirectConns` start hook, but to clarify, the ticket doesn't suggest adding a lifecycle hook

Reply:
ah sorry. Please implement this as a lifecycle hook in that case.

Reply:
the point of lifecycle hooks it to keep control and track of "processes" and "services". We want to avoid starting goroutines all over the place without some form of control.